evmrpc/info.go

```diff
@@ -229,6 +248,29 @@ func (i *InfoAPI) getRewards(block *coretypes.ResultBlock, baseFee *big.Int, rew
 	return CalculatePercentiles(rewardPercentiles, GasAndRewards, totalEVMGasUsed), nil
 }
 
+func (i *InfoAPI) getCongestionData(ctx context.Context, height *int64) (blockGasUsed uint64, err error) {
+	block, err := blockByNumber(ctx, i.tmClient, height)
+	if err != nil {
+		// block pruned from tendermint store. Skipping
+		return 0, err
+	}
+	totalEVMGasUsed := uint64(0)
+	for _, txbz := range block.Block.Txs {
+		ethtx := getEthTxForTxBz(txbz, i.txDecoder)
+		if ethtx == nil {
+			// not evm tx
+			continue
+		}
+		// okay to get from latest since receipt is immutable
+		receipt, err := i.keeper.GetReceipt(i.ctxProvider(LatestCtxHeight), ethtx.Hash())
```

Comment:
best to check if `receipt.BlockHeight` matches height in `block`, and ignore if mismatch